overlord/ifacestate/helpers.go

```diff
@@ -297,18 +297,19 @@ func CheckInterfaces(st *state.State, snapInfo *snap.Info) error {
 	// XXX: AddImplicitSlots is really a brittle interface
 	snap.AddImplicitSlots(snapInfo)
 
+	if snapInfo.SnapID == "" {
+		// no SnapID means --dangerous was given, so skip interface checks
```

Comment:
Hm, `sudo snap install --dangerous hello` has a SnapID in my state.json file. So the comment is slightly misleading (at least to me). Are we actually testing for local (sideload) installs? If so, would it make sense to check for snapInfo.Revision.Local() here?